pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -238,8 +238,13 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		metrics:                    options.Metrics,
 	}
 
-	c.serviceInformer = kubeClient.KubeInformer().Core().V1().Services().Informer()
-	c.serviceLister = kubeClient.KubeInformer().Core().V1().Services().Lister()
+	var err error
+
+	c.serviceInformer, err = kubeClient.NewKubeInformer().ForResource(v1.SchemeGroupVersion.WithResource("services"))
+	if err != nil {
```

Comment:
When will this happen?

Reply:
Only if you were to pass a resource that the factory doesn't know how to create an informer for.